install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
   printf "%s\n" "${RED}x $*${NO_COLOR}" >&2
 }
 
-complete() {
+success() {
   printf "%s\n" "${GREEN}✓${NO_COLOR} $*"
 }
 
-# Gets path to a temporary file, even if
-get_tmpfile() {
-  local suffix
-  suffix="$1"
-  if hash mktemp; then
-    printf "%s.%s" "$(mktemp)" "${suffix}"
-  else
-    # No really good options here--let's pick a default + hope
-    printf "/tmp/starship.%s" "${suffix}"
-  fi
+has() {
+  command -v "$1" >/dev/null 2>&1
 }
 
-# Test if a location is writeable by trying to write to it. Windows does not let
-# you test writeability other than by writing: https://stackoverflow.com/q/1999988
-test_writeable() {
-  local path
-  path="${1:-}/test.txt"
-  if touch "${path}" 2>/dev/null; then
-    rm "${path}"
-    return 0
-  else
+mktmpdir() {
+  tmpdir=$(mktemp -dqt "$1") || {
+    error "Can't create temp file, exiting…"
     return 1
-  fi
+  }
+  echo "$tmpdir"
 }
 
 fetch() {
-  local command
-  if hash curl 2>/dev/null; then
-    set +e
-    command="curl --silent --fail --location $1"
-    curl --silent --fail --location "$1"
-    rc=$?
-    set -e
-  else
-    if hash wget 2>/dev/null; then
-      set +e
-      command="wget -O- -q $1"
-      wget -O- -q "$1"
-      rc=$?
-      set -e
+  file="$1"
+  url="$2"
+
+  if has curl; then
+    cmd="curl -w '%{http_code}' -sL -o $file $url"
+    code=$(curl -w '%{http_code}' -sL -o "$file" "$url")
+    if [ "$code" = "200" ]; then
+      rc=0
     else
-      error "No HTTP download program (curl, wget) found…"
-      exit 1
+      rc=$code
     fi
+  elif has wget; then
+    cmd="wget -q -O $file $url"
+    {
+      wget -q -O "$file" "$url"
+      rc=$?
+    } || true
+  elif has fetch; then
```

Comment:
Doesn't this cause an infinite recursion on systems that don't have curl/wget installed? `fetch` should be defined at this point (since we're calling it), which will cause this block to infinitely succeed.

Reply:
Probably 🤦
I'll rename function to avoid that.